scapy/sendrecv.py

```diff
@@ -924,6 +924,9 @@ def _select(sockets):
             for s in ins:
                 try:
                     p = s.recv()
+                except socket.error:
+                    del sniff_sockets[s]
+                    break
```

Comment:
Ok so a few things:
- Use `continue` instead of `break` here, so that other selected sockets are processed as well
- add a `warning()` call (it’s a scapy internal function), that will also print the socket error message (you’ll need except as)

E.g
```
warning(“Socket %s failed with %s and this will be ignored” % (s, ex))
```